src/extension/providers/liquid/index.ts

```diff
@@ -0,0 +1,40 @@
+import { postMessage } from "../postMessage";
+
+declare global {
+  interface Window {
+    liquid: LiquidProvider;
+  }
+}
+
+export default class LiquidProvider {
+  enabled: boolean;
+
+  constructor() {
+    this.enabled = false;
```

Comment:
any reason to initialize to `false`? even if the user has enabled the app

Reply:
>  If the user adds a permission then it can enable without a popup. Is this what you mean?

yeah in such case, this.enabled should be `true` no ?

Reply:
No, enable() has to be called either way. It's just what happens inside enable() will change if there is a permission set.